### PR TITLE
driver/mtd/ftl: support to skip bad block for earse/bread/bwrite

### PR DESCRIPTION
**Pull request by Donny9**
## Summary
driver/mtd/ftl: support to skip bad block for earse/bread/bwrite
## Impact
1.support to use dd commond to burn romfs image to nand flash 
2.support bare-write nand flash device without file system
## Testing
daily test

### CODE REVIEW

File: drivers/mtd/ftl.c [hunk 1]
```diff
@@ -172,6 +228,163 @@ static int ftl_close(FAR struct inode *inode)
...
+      size_t count;
+
+      starteraseblock = startblock / dev->blkper;
+      if (starteraseblock >= dev->lpcount)
+        {
+          ret = -ENOSPC;
+          break;
+        }
+
+      count = ftl_get_cblock(dev, starteraseblock, (nblocks + mask) & ~mask);
```
Comment (xiaoxiang781216) [inline drivers/mtd/ftl.c:274]: @Donny9 should we divide nblocks by dev->blkper before pass to ftl_get_cblock?

File: drivers/mtd/ftl.c [hunk 2]
```diff
@@ -120,6 +126,56 @@ static const struct block_operations g_bops =
  * Private Functions
  ****************************************************************************/
 
+static int ftl_init_map(FAR struct ftl_struct_s *dev)
```
Comment (acassis) [inline drivers/mtd/ftl.c:129]: Please include the Description comment above this ftl_init_map() function

Comment (Donny9) [reply drivers/mtd/ftl.c:139]: Done~

File: drivers/mtd/ftl.c (hunk 1 above)
Comment (Donny9) [reply drivers/mtd/ftl.c:299]: yes，The parameter count of ftl get_clock indicates the number of eraseblocks, but we have already done the conversion by `(nblocks + mask) & ~mask)`

**Pull request merged**